code/LeftAndMain.php

```diff
@@ -867,6 +867,9 @@ public function redirect($url, $code = 302)
             if ($this->getRequest()->getHeader('X-Pjax') && !$response->getHeader('X-Pjax')) {
                 $response->addHeader('X-Pjax', $this->getRequest()->getHeader('X-Pjax'));
             }
+            if ($this->getRequest()->getHeader('X-Status') && !$response->getHeader('X-Status')) {
+                $response->addHeader('X-Status', $this->getRequest()->getHeader('X-Status'));
+            }
```

Comment:
What is the intention here? I can't think of a scenario where the _request_ would ever include `X-Status`?

Reply:
We need this since we lose these headers (X-Pjax and X-Status) when we sent request for redirection from `GridFieldDetailForm_ItemRequest::redirectAfterSave`.

Reply:
Can you please add a comment about that to the codebase directly? It could help explain what's happening if someone comes across that code in the future.

Reply:
Updated